_stbt/core.py

```diff
@@ -1913,8 +1922,7 @@ def _push_sample(self, sample):
         annotations = []
         with self.annotations_lock:
             for x in self.text_annotations:
-                x.setdefault('start_time', now)
-                if now < x['start_time'] + x['duration']:
+                if now < x.end_time:
                     texts.append(x)
             self.text_annotations = texts[:]
```

Comment:
Now we don't have setdefault this could be "simplified" to a list comprehension:

    # Remove expired annotations
    self.text_annotations = [x for x in self.text_annotations if now < x.end_time]
    current_annotations = [x for x in self.text_annotations if x.time <= now]

Not important

Reply:
Good idea, done.